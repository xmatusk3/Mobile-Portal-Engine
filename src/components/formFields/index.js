export * from './TextArea';
export * from './TextBox';
export * from './DropDownList';
export * from './MultipleChoice';
export * from './LabelText';
export * from './NotSupportedField';
export * from './SingleChoice';
export * from './Checkbox';
export * from './RadioButtons';
