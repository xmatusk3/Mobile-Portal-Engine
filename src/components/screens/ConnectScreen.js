import React, { Component } from 'react';
import { Text, View } from 'react-native';
import { Button } from 'react-native-elements';
import { connect } from 'react-redux';

import { Card, CardSection, Input } from '../common';
import { LoadingScreen } from '.';
import { toggleLoading, setServerAddress, setError, connectToServer } from '../../actions';

class ConnectScreen extends Component {
  _onConnect = () => {
    this.props.setError('');
    if (!this.props.address) {
      this.props.setError('Address field cannot be empty!');
      return;
    }

    this.props.toggleLoading();
    this.props.connectToServer(this.props.address, this.props.navigation.navigate);
  };

  _renderConnect = () => {
    const {
      containerStyle,
      errorTextStyle,
      buttonCardSectionStyle,
      boldTextStyle,
      normalTextStyle,
      buttonStyle,
    } = styles;

    return (
      <View style={containerStyle}>
        <View>
          <Text style={boldTextStyle}>Kentico</Text>
          <Text style={normalTextStyle}>Pages</Text>
        </View>

        <Card>
          <CardSection>
            <Input
              label={'Address:'}
              placeholder={'127.0.0.1'}
              value={this.props.address}
              onChangeText={text => this.props.setServerAddress(text)}
            />
          </CardSection>

          <Text style={errorTextStyle}>{this.props.error}</Text>

          <CardSection style={buttonCardSectionStyle}>
            <Button title="Connect" buttonStyle={buttonStyle} onPress={this._onConnect} raised />
          </CardSection>
        </Card>
      </View>
    );
  };

  render() {
    if (this.props.loading) {
      return <LoadingScreen />;
    }

    return this._renderConnect();
  }
}

const styles = {
  containerStyle: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#ededed',
  },
  buttonStyle: {
    backgroundColor: '#497d04',
    shadowColor: '#355e00',
    shadowOffset: { height: -3, width: 0 },
  },
  buttonCardSectionStyle: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  errorTextStyle: {
    fontSize: 20,
    alignSelf: 'center',
    color: 'red',
  },
  boldTextStyle: {
    fontSize: 45,
    fontWeight: 'bold',
    textAlign: 'center',
    color: '#2a2a2a',
  },
  normalTextStyle: {
    fontSize: 40,
    textAlign: 'center',
    color: '#2a2a2a',
    marginBottom: 50,
    marginTop: 10,
  },
};

const mapStateToProps = ({ global: { loading, error }, auth: { address } }) => {
  return { loading, error, address };
};

const connectedComponent = connect(mapStateToProps, {
  toggleLoading,
  setServerAddress,
  setError,
  connectToServer,
})(ConnectScreen);

export { connectedComponent as ConnectScreen };
