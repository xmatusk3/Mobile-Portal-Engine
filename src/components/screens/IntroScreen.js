import React, { Component } from 'react';
import _ from 'lodash';

import { View, Text, AsyncStorage } from 'react-native';

import Slides from '../common/Slides';
import { LoadingScreen } from '.';

const SLIDES_TEXT = [
  'Welcome to Kentico Pages!',
  'Use this to edit your site content',
  'Swipe from the left to access specific page and start customizing',
  'But first enter your Kentico configuration to access your server!',
];

class IntroScreen extends Component {
  state = { token: null };

  async componentWillMount() {
    let token = await AsyncStorage.getItem('kentico_complete_intro');

    if (token) {
      this.props.navigation.navigate('connect');
    } else {
      this.setState({ token: false });
    }
  }

  _onSlideComplete = () => {
    this.props.navigation.navigate('connect');
  };

  render() {
    if (_.isNull(this.state.token)) {
      return <LoadingScreen />;
    }

    return (
      <View style={{ flex: 1 }}>
        <Slides data={SLIDES_TEXT} onComplete={this._onSlideComplete} />
      </View>
    );
  }
}

export { IntroScreen };
