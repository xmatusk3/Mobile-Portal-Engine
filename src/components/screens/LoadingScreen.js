import React, { Component } from 'react';
import { View, Text, Dimensions, ActivityIndicator } from 'react-native';

const SCREEN_WIDTH = Dimensions.get('window').width;

class LoadingScreen extends Component {
  render() {
    return (
      <View style={styles.containerStyle}>
        <View>
          <Text style={styles.boldStyle}>Kentico</Text>
          <Text style={styles.normalStyle}>Pages</Text>
        </View>
        <ActivityIndicator size="large" style={styles.loaderStyle} />
      </View>
    );
  }
}

const styles = {
  boldStyle: {
    fontSize: 45,
    fontWeight: 'bold',
    textAlign: 'center',
    color: '#2a2a2a',
  },
  normalStyle: {
    fontSize: 40,
    textAlign: 'center',
    color: '#2a2a2a',
    marginTop: 10,
    marginBottom: 50,
  },
  containerStyle: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: SCREEN_WIDTH,
    backgroundColor: '#ededed',
  },
  loaderStyle: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 150,
  },
};

export { LoadingScreen };
