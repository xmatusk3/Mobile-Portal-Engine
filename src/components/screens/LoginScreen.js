import React, { Component } from 'react';
import { View, Text, ActivityIndicator } from 'react-native';
import { Button } from 'react-native-elements';
import { connect } from 'react-redux';

import { Input, Card, CardSection } from '../common';
import { LoadingScreen } from '.';
import { setError, toggleLoading, authorizeToServer } from '../../actions';

class LoginScreen extends Component {
  state = { login: '', password: '' };

  componentWillMount() {
    if (this.props.loading) {
      this.props.toggleLoading();
    }
  }

  _onLogin = () => {
    this.props.setError('');
    if (!this.state.login) {
      this.props.setError('Login field cannot be empty!');
      return;
    }

    this.props.toggleLoading();
    this.props.authorizeToServer(this.state.login, this.state.password, this.props.navigation.navigate);
  };

  _renderLogin = () => {
    const {
      containerStyle,
      errorTextStyle,
      buttonCardSectionStyle,
      boldTextStyle,
      normalTextStyle,
      buttonStyle,
    } = styles;

    return (
      <View style={containerStyle}>
        <View>
          <Text style={boldTextStyle}>Kentico</Text>
          <Text style={normalTextStyle}>Pages</Text>
        </View>

        <Card>
          <CardSection>
            <Input
              label={'Login:'}
              placeholder={'user name'}
              value={this.state.login}
              onChangeText={text => this.setState({ login: text })}
            />
          </CardSection>

          <CardSection>
            <Input
              label={'Password:'}
              placeholder={'password'}
              value={this.state.password}
              onChangeText={text => this.setState({ password: text })}
              secureTextEntry
            />
          </CardSection>

          <Text style={errorTextStyle}>{this.props.error}</Text>

          <CardSection style={buttonCardSectionStyle}>
            <Button title="Login" buttonStyle={buttonStyle} onPress={this._onLogin} raised />
          </CardSection>
        </Card>
      </View>
    );
  };

  render() {
    if (this.props.loading) {
      return <LoadingScreen />;
    }

    return this._renderLogin();
  }
}

const styles = {
  containerStyle: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#ededed',
  },
  buttonStyle: {
    backgroundColor: '#497d04',
    shadowColor: '#355e00',
    shadowOffset: { height: -3, width: 0 },
  },
  buttonCardSectionStyle: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  errorTextStyle: {
    fontSize: 20,
    alignSelf: 'center',
    color: 'red',
  },
  boldTextStyle: {
    fontSize: 45,
    fontWeight: 'bold',
    textAlign: 'center',
    color: '#2a2a2a',
  },
  normalTextStyle: {
    fontSize: 40,
    textAlign: 'center',
    color: '#2a2a2a',
    marginBottom: 50,
    marginTop: 10,
  },
};

const mapStateToProps = ({ global: { loading, error } }) => {
  return { loading, error };
};

const connectedComponent = connect(mapStateToProps, {
  setError,
  toggleLoading,
  authorizeToServer,
})(LoginScreen);

export { connectedComponent as LoginScreen };
