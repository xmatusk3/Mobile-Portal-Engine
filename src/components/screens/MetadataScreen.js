import React, { Component } from 'react';
import { Picker, Text, View, ScrollView } from 'react-native';
import { Button } from 'react-native-elements';
import { connect } from 'react-redux';
import _ from 'lodash';

import { Card, CardSection, Input } from '../common';
import { fetchMetadata, updateMetadata, toggleLoading } from '../../actions';
import { LoadingScreen } from '.';

class MetadataScreen extends Component {
  state = {
    pageTitle: '',
    pageDescription: '',
    pageKeywords: '',
    pageTags: '',
    selectedTagGroup: '',
  };

  componentWillMount() {
    this.props.fetchMetadata();
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.pageId !== nextProps.pageId) {
      this.props.fetchMetadata();
    }

    if (!nextProps.metadata) return;

    this.setState({
      pageTitle: nextProps.metadata.documentTitle,
      pageDescription: nextProps.metadata.documentDescription,
      pageKeywords: nextProps.metadata.documentKeywords,
      pageTags: nextProps.metadata.documentTags,
      selectedTagGroup: `${Object.keys(nextProps.tagGroups)[0]}`
    });
  }

  _createPickerItems = () => {
    return Object.values(this.props.tagGroups)
      .map(group => 
        <Picker.Item key={group.tagGroupID} label={group.tagGroupDisplayName} value={group.tagGroupID} />
      );
  }

  _onSave = () => {
    this.props.toggleLoading();
    
    this.props.updateMetadata(
      this.state.pageTitle,
      this.state.pageDescription,
      this.state.pageKeywords,
      this.state.selectedTagGroup,
      this.state.pageTags
    );
  }

  render() {
    if (this.props.loading || !this.props.metadata) {
      return <LoadingScreen />;
    }

    return (
      <View style={{ flex: 1 }}>
        <View style={{ flex: 10 }}>
          <ScrollView style={styles.containerStyle}>
            <Card>
              <CardSection style={styles.sectionStyle}>
                <Text style={styles.boldStyle}>Page settings</Text>
                <Input 
                  label={'Page title:'}
                  placeholder={'Page title'}
                  value={this.state.pageTitle}
                  onChangeText={text => this.setState({ pageTitle: text })}
                  labelCustomStyle={styles.labelStyle}
                  containerCustomStyle={styles.singleRowContainerStyle}
                  inputCustomStyle={styles.singleRowInputStyle}
                />

                <Input 
                  label={'Page description:'}
                  placeholder={'Page description'}
                  value={this.state.pageDescription}
                  multiline
                  onChangeText={text => this.setState({ pageDescription: text })}
                  labelCustomStyle={styles.labelStyle}
                  containerCustomStyle={styles.multipleRowsContainerStyle}
                  inputCustomStyle={styles.multipleRowsInputStyle}
                />

                <Input 
                  label={'Page keywords (separated by comma):'}
                  placeholder={'Cafe, Coffee, ...'}
                  value={this.state.pageKeywords}
                  multiline
                  onChangeText={text => this.setState({ pageKeywords: text })}
                  labelCustomStyle={styles.labelStyle}
                  containerCustomStyle={styles.multipleRowsContainerStyle}
                  inputCustomStyle={styles.multipleRowsInputStyle}
                />
              </CardSection>
            </Card>

            {/* ---------TAGS!-----------*/}

            {this.props.tagGroups && !_.isEmpty(this.props.tagGroups) &&
            <Card>
              <CardSection style={styles.sectionStyle}>
                <Text style={styles.boldStyle}>Tags</Text>
                <Text style={styles.normalStyle}>Page tag group:</Text>
                <View style={styles.pickerContainerStyle}>
                  <Picker
                    selectedValue={this.state.selectedTagGroup}
                    style={styles.pickerStyle}
                    onValueChange={itemValue => this.setState({ selectedTagGroup: itemValue })}
                  >
                    {this._createPickerItems()}
                  </Picker>
                </View>

                <Input 
                  label={'Page tags (separated by comma):'}
                  placeholder={'dogs, "angry birds", cats'}
                  value={this.state.pageTags}
                  multiline
                  onChangeText={text => this.setState({ pageTags: text })}
                  labelCustomStyle={styles.labelStyle}
                  containerCustomStyle={styles.multipleRowsContainerStyle}
                  inputCustomStyle={styles.multipleRowsInputStyle}
                />
              </CardSection>
            </Card>
            }
          </ScrollView>
        </View>
        <View style={{ flex: 1 }}>
          <Button
            title='Save'
            buttonStyle={styles.buttonStyle}
            onPress={this._onSave}
            containerViewStyle={styles.buttonContainerStyle}
          />
        </View>
      </View>
    );
  }
}

const styles = {
  buttonStyle: {
    backgroundColor: '#497d04', 
    flex: 1,
  },
  buttonContainerStyle: {
    width: '100%',
    marginLeft: 0, 
    flex: 1
  },
  boldStyle: {
    fontSize: 26,
    fontWeight: 'bold',
    textAlign: 'left',
    color: '#2a2a2a',
  },
  normalStyle: {
    fontSize: 18,
    marginTop: 5,
  },
  containerStyle: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#ededed',
  },
  sectionStyle: {
    flexDirection: 'column'
  },
  labelStyle: {
    paddingLeft: 0,
    marginTop: 5,
  },
  singleRowInputStyle: {
    marginTop: 5,
    marginLeft: 3,
    height: 40,
    flex: 1,
    borderRadius: 7,
    borderWidth: 0.5,
    borderColor: '#bbb'
  },
  singleRowContainerStyle: { 
    height: 40,
    width: '100%'
  },
  multipleRowsContainerStyle: {
    flexDirection: 'column',
    alignItems: 'flex-start',
    height: 140,
    marginBottom: 5
  },
  multipleRowsInputStyle: {
    textAlignVertical: 'top',
    height: 100,
    flex: 1,
    borderWidth: 0.5,
    borderRadius: 7,
    borderColor: '#bbb',
    width: '100%',
  },
  pickerStyle: {
    width: '100%',
  },
  pickerContainerStyle: {
    borderRadius: 7,
    borderWidth: 0.5,
    borderColor: '#bbb'
  }
};

const mapStateToProps = ({ selectedItem, auth, global }) => {
  return {
    pageId: selectedItem.documentID,
    metadata: selectedItem.metadata,
    tagGroups: auth.selectedSite.tagGroups,
    loading: global.loading
  };
};

const connectedComponent = connect(mapStateToProps, { fetchMetadata, updateMetadata, toggleLoading })(MetadataScreen);

export { connectedComponent as MetadataScreen };
