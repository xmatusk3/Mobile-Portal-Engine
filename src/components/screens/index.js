export * from './IntroScreen';
export * from './LoadingScreen';
export * from './LoginScreen';
export * from './PreviewScreen';
export * from './ConnectScreen';
export * from './SetSiteScreen';
export * from './WorkflowScreen';
export * from './MetadataScreen';
