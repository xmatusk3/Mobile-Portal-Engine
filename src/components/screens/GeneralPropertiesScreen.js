import React, { Component } from 'react';
import { Picker, Text, View, ScrollView } from 'react-native';
import { Button } from 'react-native-elements';
import { connect } from 'react-redux';
import _ from 'lodash';

import { Card, CardSection } from '../common';
import { TextBox, TextArea, DropDownList, MultipleChoice } from '../formFields';
import { fetchGeneralProperties } from '../../actions';
import { LoadingScreen, NoReadPermissionScreen } from '.';
import utils from '../../utils/utils';

class GeneralPropertiesScreen extends Component {
  state = {
    serverMessage: '',
    serverMessageIsError: false,
  };

  componentWillMount() {
    this.props.navigation.addListener('willFocus', this.props.fetchGeneralProperties);
  }

  componentWillUnmount() {
    this.props.navigation.removeListener('willFocus', this.props.fetchGeneralProperties);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.hasModifyPermission === false) {
      this.setState({
        serverMessage: 'You cannot modify this page.',
        serverMessageIsError: false
      });
    }
    if (nextProps.properties) {
      Object.values(nextProps.properties)
        .map(propSection => Object.keys(propSection)
          .map(propKey => this._setStateForProp(propSection[propKey], propKey)));
    }
  }

  _setStateForProp = (property, key) => {
    let value = property;
    let stateValue;
    let inheritValue;

    while (value) {
      if (value.fieldSelectedValueID) {
        stateValue = value.fieldSelectedValueID;    
      }

      if (value.checkboxValue !== undefined) {
        inheritValue = value.checkboxValue;
      }

      if (value.id !== undefined) {
        stateValue = value.fieldValue;
      }

      if (value.selectedIndex !== undefined) {
        stateValue = value.selectedIndex;
      }

      if (value.fieldValue === undefined) {
        break;
      }

      value = value.fieldValue;
    }

    stateValue = stateValue !== undefined ? stateValue : value;
    // Check for date value
    if (typeof(stateValue) === "string") {
      stateValue = isNaN(new Date(stateValue)) ? stateValue : new Date(stateValue);
    }

    if (inheritValue !== undefined) {
      this.setState({ [key]: stateValue, [`${key}Inherit`]: inheritValue });
    } else {
      this.setState({ [key]: stateValue });
    }
  }

  _createResetRatingButton = (ratingField) => {
    return (
      <View key={'rating'}>
        {ratingField}
        <Button
          title='Reset rating'
          color='#262524'
          disabled={!this.props.hasModifyPermission}
          buttonStyle={styles.buttonStyle}
          onPress={() => console.log('..press..')}
          containerViewStyle={[styles.buttonContainerStyle, { width: 120 }]}
        />
      </View>
    )
  }

  _createClearOutputCacheButton = (cacheMinutesField) => {
    return (
      <View key={'outputCache'}>
        {cacheMinutesField}
        <Button
          title='Clear output cache'
          color='#262524'
          disabled={!this.props.hasModifyPermission}
          buttonStyle={styles.buttonStyle}
          onPress={() => console.log('..pressCache..')}
          containerViewStyle={[styles.buttonContainerStyle, { width: 160 }]}
        />
      </View>
    );
  }

  _createOwnerButtons = (ownerField) => {
    return (
      <View key={'owner'}>
        {ownerField}
        <View style={styles.ownerButtonContainer}>
          <Button
            title='Select'
            color='#262524'
            disabled={!this.props.hasModifyPermission}
            buttonStyle={styles.buttonStyle}
            onPress={() => console.log('..pressSelectOwner..')}
            containerViewStyle={styles.buttonContainerStyle}
          />
          <Button
            title='Clear'
            color='#262524'
            disabled={!this.props.hasModifyPermission}
            buttonStyle={styles.buttonStyle}
            onPress={() => console.log('..pressClearOwner..')}
            containerViewStyle={styles.buttonContainerStyle}
          />
        </View>
      </View>
    );
  }

  _createOwnedByGroupButton = (ownedByGroupField) => {
    return (
      <View key={'ownedByGroup'}>
        {ownedByGroupField}
        <Button
          title='Change'
          color='#262524'
          disabled={!this.props.hasModifyPermission}
          buttonStyle={styles.buttonStyle}
          onPress={() => console.log('..pressChangeOwnedByGroup..')}
          containerViewStyle={styles.buttonContainerStyle}
        />
      </View>
    );
  }

  _renderDesignSection = () => {
    const { properties: { designGeneralProperties }, hasModifyPermission, pageIsNotRoot } = this.props;

    return Object.keys(designGeneralProperties)
      .map((key, index) => {
        const object = designGeneralProperties[key];

        return utils.generateFormField({
          fieldObject: object, 
          value: this.state[key],
          onChange: (value) => this.setState({ [key]: value }),
          disabled: !hasModifyPermission || this.state[`${key}Inherit`],
          isRoot: !pageIsNotRoot,
          checkboxValue: this.state[`${key}Inherit`],
          onCheckboxChange: () => this.setState({ [`${key}Inherit`]: !this.state[`${key}Inherit`]}),
          checkboxDisabled: !hasModifyPermission,
          index
        });
      });
  }

  _renderOtherSection = () => {
    const { properties: { otherGeneralProperties }, pageIsNotRoot } = this.props;

    return Object.keys(otherGeneralProperties)
      .map((key, index) => {
        let object = otherGeneralProperties[key];

        if (!pageIsNotRoot && object.fieldValue === null) {
          return null;
        }

        if (!pageIsNotRoot && object.fieldValue === '') {
          object = { ...object, fieldValue: '(root)' };
        }

        if (key === 'rating') {
          return this._createResetRatingButton(utils.generateFormField({ fieldObject: object, value: this.state[key], index }));
        }

        return utils.generateFormField({ fieldObject: object, value: this.state[key], index });
      });
  }

  _renderOwnerSection = () => {
    const { properties: { ownerGeneralProperties } } = this.props;

    return Object.keys(ownerGeneralProperties)
      .map((key, index) => {
        const object = ownerGeneralProperties[key];

        if (key === 'owner') {
          return this._createOwnerButtons(utils.generateFormField({ fieldObject: object, value: this.state[key], index }));
        }
        else {
          return this._createOwnedByGroupButton(utils.generateFormField({ fieldObject: object, value: this.state[key], index }));
        }
      });
  }

  _renderOutputCacheSection = () => {
    const { properties: { outputCacheGeneralProperties }, hasModifyPermission } = this.props;

    return Object.keys(outputCacheGeneralProperties)
      .map((key, index) => {
        const object = outputCacheGeneralProperties[key];

        if (key === 'cacheMinutes') {
          return this._createClearOutputCacheButton(utils.generateFormField({
            fieldObject: object,
            value: this.state[key],
            onChange: (value) => this.setState({ [key]: value }),
            disabled: !hasModifyPermission || this.state.useOutputCache !== 0,
            index
          }));
        }

        return utils.generateFormField({
          fieldObject: object,
          value: this.state[key],
          onChange: (value) => this.setState({ [key]: value }),
          disabled: !hasModifyPermission,
          index
        });
      })
  }

  
  _renderOnlineMarketingSection = () => {
    const { properties: { onlineMarketingGeneralProperties }, hasModifyPermission, pageIsNotRoot } = this.props;

    return Object.keys(onlineMarketingGeneralProperties)
      .map((key, index) => {
        const object = onlineMarketingGeneralProperties[key];

        return utils.generateFormField({
          fieldObject: object,
          value: this.state[key],
          onChange: () => this.setState({ [key]: !this.state[key] }),
          disabled: !hasModifyPermission || this.state[`${key}Inherit`],
          isRoot: !pageIsNotRoot,
          checkboxValue: this.state[`${key}Inherit`],
          onCheckboxChange: () => this.setState({ [`${key}Inherit`]: !this.state[`${key}Inherit`] }),
          checkboxDisabled: !hasModifyPermission,
          index
        });
      })
  }

  render() {
    if (this.props.loading) {
      return <LoadingScreen />
    }

    if (this.props.hasReadPermission === undefined) {
      return <NoReadPermissionScreen />
    }

    const { serverMessage, serverMessageIsError, hasReadPermission } = this.props;
    
    return (
      <View style={{flex: 1}}>
      {serverMessage ? 
        <View style={styles.serverMessageContainer}>
          <Text style={serverMessageIsError ? styles.errorText : styles.successText}>
            {serverMessage}
          </Text> 
        </View>
      : null}
        <View style={{ flex: serverMessage ? 9.5 : 10 }}>
          <ScrollView style={styles.containerStyle}>
            <Card>
              <CardSection style={styles.sectionStyle}>
                <Text style={styles.boldStyle}>Design</Text>
                {this._renderDesignSection()}
              </CardSection>
            </Card>

            <Card>
              <CardSection style={styles.sectionStyle}>
                <Text style={styles.boldStyle}>Other properties</Text>
                {this._renderOtherSection()}
              </CardSection>
            </Card>

            <Card>
              <CardSection style={styles.sectionStyle}>
                <Text style={styles.boldStyle}>Owner</Text>
                {this._renderOwnerSection()}
              </CardSection>
            </Card>

            <Card>
              <CardSection style={styles.sectionStyle}>
                <Text style={styles.boldStyle}>Output cache</Text>
                {this._renderOutputCacheSection()}
              </CardSection>
            </Card>

            <Card>
              <CardSection style={styles.sectionStyle}>
                <Text style={styles.boldStyle}>On-line marketing</Text>
                {this._renderOnlineMarketingSection()}
              </CardSection>
            </Card>
          </ScrollView>
        </View>
      </View>
    );
  }
}

const styles = {
  ownerButtonContainer: {
    flexDirection: 'row',
  },
  buttonStyle: {
    backgroundColor: '#bdbbbb',
    flex: 1,
    borderWidth: 1,
    marginLeft: 5
  },
  buttonContainerStyle: {
    width: 80,
    height: 35,
    marginLeft: 0,
  },
  saveButtonStyle: {
    backgroundColor: '#497d04', 
    flex: 1,
  },
  saveButtonContainerStyle: {
    width: '100%',
    marginLeft: 0, 
    flex: 1
  },
  boldStyle: {
    fontSize: 26,
    fontWeight: 'bold',
    textAlign: 'left',
    color: '#2a2a2a',
  },
  containerStyle: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#ededed',
  },
  sectionStyle: {
    flexDirection: 'column'
  },
  serverMessageContainer: {
    flex: 0.5,
    justifyContent: 'center',
    alignItems: 'center',
  },
  errorText: {
    fontSize: 20,
    alignSelf: 'center',
    color: 'red',
  },
  successText: {
    fontSize: 20,
    alignSelf: 'center',
  }
};

const mapStateToProps = ({ selectedItem, global }) => {
  if (!selectedItem.generalProperties) return { loading: true }

  const { 
    designGeneralProperties, 
    otherGeneralProperties, 
    onlineMarketingGeneralProperties, 
    ownerGeneralProperties, 
    outputCacheGeneralProperties, 
    hasModifyPermission, 
    hasReadPermission
  } = selectedItem.generalProperties;

  return {
    properties: {
      designGeneralProperties, 
      otherGeneralProperties, 
      onlineMarketingGeneralProperties, 
      ownerGeneralProperties, 
      outputCacheGeneralProperties,
    }, 
    hasModifyPermission, 
    hasReadPermission,
    loading: global.loading,
    pageIsNotRoot: !!selectedItem.documentName,
  };
};

const connectedComponent = connect(mapStateToProps, { fetchGeneralProperties })(GeneralPropertiesScreen);

export { connectedComponent as GeneralPropertiesScreen };
