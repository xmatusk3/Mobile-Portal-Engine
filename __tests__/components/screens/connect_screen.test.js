import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import { ConnectScreen } from '../../../src/components/screens';
import store from '../../../src/store';

describe('ConnectScreen component', () => {
  it('renders correctly', () => {
    const connect = renderer
      .create(
        <Provider store={store}>
          <ConnectScreen />
        </Provider>
      )
      .toJSON();
    expect(connect).toMatchSnapshot();
  });
});
