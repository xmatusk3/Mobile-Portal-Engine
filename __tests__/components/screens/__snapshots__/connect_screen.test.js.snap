// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectScreen component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ededed",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2a2a2a",
          "fontSize": 45,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Kentico
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2a2a2a",
          "fontSize": 40,
          "marginBottom": 50,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      Pages
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#ddd",
        "borderRadius": 2,
        "borderWidth": 1,
        "elevation": 1,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ededed",
            "borderBottomWidth": 1,
            "borderColor": "#ddd",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "maxWidth": 450,
            "padding": 5,
            "position": "relative",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 18,
                "paddingLeft": 20,
              },
              undefined,
            ]
          }
        >
          Address:
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="127.0.0.1"
          secureTextEntry={undefined}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 18,
                "height": 40,
                "lineHeight": 23,
                "paddingLeft": 5,
                "paddingRight": 5,
                "width": 200,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value="matusmkenticopages.azurewebsites.net"
        />
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "color": "red",
          "fontSize": 20,
        }
      }
    >
      
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ededed",
            "borderBottomWidth": 1,
            "borderColor": "#ddd",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "maxWidth": 450,
            "padding": 5,
            "position": "relative",
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            undefined,
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#497d04",
                  "shadowColor": "#355e00",
                  "shadowOffset": Object {
                    "height": -3,
                    "width": 0,
                  },
                },
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Connect
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
